File: ruby/yapay_gw/lib/yapay_gw/models/oneclick_register_data.rb
class OneClickRegisterData
    attr_accessor :store_code
    attr_accessor :card_holder_name
    attr_accessor :card_number
    attr_accessor :expiration_date
    attr_accessor :buyer_email
    attr_accessor :payment_code

    def as_json(options = {})
        json = {:codigoEstabelecimento => store_code, 
                :nomeTitularCartaoCredito => card_holder_name,
                :numeroCartaoCredito => card_number,
                :dataValidadeCartao => expiration_date,
                :emailComprador => buyer_email,
                :formaPagamento => payment_code
        }
        json.delete_if { |k, v| v.nil? }
        json
    end
end

File: dotnet/YapayGatewayLib/model/OneClickRegisterData.cs
using System;
using Newtonsoft.Json;

namespace YapayGatewayLib.Model
{
    public class OneClickRegisterData
    {
       [JsonProperty(PropertyName = "codigoEstabelecimento")]
       public string storeCode { get; set; }

       [JsonProperty(PropertyName = "nomeTitularCartaoCredito")]
       public string cardHolderName { get; set; }

       [JsonProperty(PropertyName = "numeroCartaoCredito")]
       public string cardNumber { get; set; }

       [JsonProperty(PropertyName = "dataValidadeCartao")]
       public string expirationDate { get; set; }

       [JsonProperty(PropertyName = "emailComprador")]
       public string buyerEmail { get; set; }

       [JsonProperty(PropertyName = "formaPagamento")]
       public long paymentCode { get; set; }
    }
}
